cumulusci/tasks/robotframework/robotperflistener.py

```diff
@@ -0,0 +1,149 @@
+import os
+import json
+import time
+
+
+def mean(numbers):
+    return float(sum(numbers)) / max(len(numbers), 1)
+
+
+class TestMetrics:
+    """Manage both REST API (predefined) metrics and also custom metrics"""
+
+    def __init__(self):
+        self.api_metrics = []
+        self.custom_metrics = {}
+
+    def summarize(self):
+        collections = {}
+        for metricset in self.api_metrics:
+            assert isinstance(metricset, dict), metricset
+            for name, value in metricset.items():
+                if not name.startswith("_"):
+                    collections.setdefault(name, []).append(float(value))
```

Comment:
You could initialize collections as a defaultdict(list) and then this would simply be `collections[name].append(float(value))`

Reply:
True, and I did think about that. But to me they are potato/potato and the other way requires an extra import.

But then from my point of view they are both equally idiomatic and perhaps newer Python programmers consider setdefault archaic for this use-case.